app/server.py

```diff
@@ -63,6 +63,8 @@ def search(search_engine):
         if db['queries'].find({query: query}).limit(1) is False:
             db['queries'].insert(
                 {"query": query, "engine": engine, "qformat": qformat})
+            db['queries'].save(
+                {"query": query, "engine": engine, "qformat": qformat})
```

Comment:
My sense is that all three of these elements are contained in the URL.  Why not just store the URL?  We also need to store the links that come back from the query plus a timestamp.

Reply:
One more thing, the url contains the number of results to be fetched, this is one extra thing, but we can work on that later, let's not make this PR too complex